Tasks in this notebook:  
* Importing datasets for data cleaning.  
* Saving the clean datasets to work with them later.

In [20]:
#Importing  libraries
import pandas as pd
import numpy as np

In [21]:
#Importing datasets
data_co2 = pd.read_csv('/Users/anna/data/climate-change/datasets/carbondioxide.csv')
data_temp = pd.read_csv('/Users/anna/data/climate-change/datasets/GlobalTemperatures.csv')
data_sea = pd.read_csv('/Users/anna/data/climate-change/datasets/seaice.csv')

## Carbon dioxide

First of all, we have a look at the data to see the information we have and its main characteristics.

In [22]:
data_co2.head()

Year  Month  Decimal Date  Carbon Dioxide (ppm)  \
0  1958      1     1958.0411                   NaN   
1  1958      2     1958.1260                   NaN   
2  1958      3     1958.2027                315.69   
3  1958      4     1958.2877                317.45   
4  1958      5     1958.3699                317.50   

   Seasonally Adjusted CO2 (ppm)  Carbon Dioxide Fit (ppm)  \
0                            NaN                       NaN   
1                            NaN                       NaN   
2                         314.42                    316.18   
3                         315.15                    317.30   
4                         314.73                    317.83   

   Seasonally Adjusted CO2 Fit (ppm)  
0                                NaN  
1                                NaN  
2                             314.89  
3                             314.98  
4                             315.06

In [23]:
data_co2.dtypes

Year                                   int64
Month                                  int64
Decimal Date                         float64
Carbon Dioxide (ppm)                 float64
Seasonally Adjusted CO2 (ppm)        float64
Carbon Dioxide Fit (ppm)             float64
Seasonally Adjusted CO2 Fit (ppm)    float64
dtype: object

In [24]:
data_co2.isna().sum()

Year                                  0
Month                                 0
Decimal Date                          0
Carbon Dioxide (ppm)                 17
Seasonally Adjusted CO2 (ppm)        17
Carbon Dioxide Fit (ppm)             13
Seasonally Adjusted CO2 Fit (ppm)    13
dtype: int64

In [25]:
data_co2.shape

(720, 7)

The types of the columns seem to be all correct.  
We have a few null values, and since they are not a lot (less than 20 out of 720), we weill just drop them.  
Looking at the last five columns, we will only use the last one, 'Seasonally Adjusted CO2 Fit (ppm)'.

In [26]:
# Renaming columns and selecting the ones we want
data_co2.rename(columns={'Seasonally Adjusted CO2 Fit (ppm)':'CO2'}, inplace=True)
co2 = data_co2[['Year', 'Month', 'CO2']]
co2.head()

Year  Month     CO2
0  1958      1     NaN
1  1958      2     NaN
2  1958      3  314.89
3  1958      4  314.98
4  1958      5  315.06

In [27]:
#Dropping NaNs
co2 = co2.dropna().reset_index(drop=True)
co2.head()

Year  Month     CO2
0  1958      3  314.89
1  1958      4  314.98
2  1958      5  315.06
3  1958      6  315.14
4  1958      7  315.21

In [28]:
#Saving clean dataset
co2.to_csv('/Users/anna/data/climate-change/datasets/clean_co2.csv', index=False)

## Temperature

Repeating the process of checking the information of the dataset, types of columns and NaNs.

In [29]:
data_temp.head()

dt  LandAverageTemperature  LandAverageTemperatureUncertainty  \
0  1750-01-01                   3.034                              3.574   
1  1750-02-01                   3.083                              3.702   
2  1750-03-01                   5.626                              3.076   
3  1750-04-01                   8.490                              2.451   
4  1750-05-01                  11.573                              2.072   

   LandMaxTemperature  LandMaxTemperatureUncertainty  LandMinTemperature  \
0                 NaN                            NaN                 NaN   
1                 NaN                            NaN                 NaN   
2                 NaN                            NaN                 NaN   
3                 NaN                            NaN                 NaN   
4                 NaN                            NaN                 NaN   

   LandMinTemperatureUncertainty  LandAndOceanAverageTemperature  \
0                            NaN                             NaN   
1                            NaN                             NaN   
2                            NaN                             NaN   
3                            NaN                             NaN   
4                            NaN                             NaN   

   LandAndOceanAverageTemperatureUncertainty  
0                                        NaN  
1                                        NaN  
2                                        NaN  
3                                        NaN  
4                                        NaN

In [30]:
data_temp.shape

(3192, 9)

In [31]:
# Selecting columns we need and renaming
temp = data_temp.rename(columns = {'dt':'Year', 'LandAndOceanAverageTemperature':'AvgTemp'})
temp = temp[['Year', 'AvgTemp']]
temp.head()

Year  AvgTemp
0  1750-01-01      NaN
1  1750-02-01      NaN
2  1750-03-01      NaN
3  1750-04-01      NaN
4  1750-05-01      NaN

In [32]:
#Dropping NaNs
temp = temp.dropna().reset_index(drop=True)
temp.head()

Year  AvgTemp
0  1850-01-01   12.833
1  1850-02-01   13.588
2  1850-03-01   14.043
3  1850-04-01   14.667
4  1850-05-01   15.507

In [33]:
temp.isna().sum()

Year       0
AvgTemp    0
dtype: int64

In [34]:
# Changing Year dtype 
temp.dtypes

Year        object
AvgTemp    float64
dtype: object

In [35]:
temp['Year'] = pd.to_datetime(temp['Year'])
temp.dtypes

Year       datetime64[ns]
AvgTemp           float64
dtype: object

In [36]:
temp.head()

Year  AvgTemp
0 1850-01-01   12.833
1 1850-02-01   13.588
2 1850-03-01   14.043
3 1850-04-01   14.667
4 1850-05-01   15.507

In [37]:
#Saving clean dataset
temp.to_csv('/Users/anna/data/climate-change/datasets/clean_temp.csv', index=False)

## Sea ice

Repeating the process of checking the information of the dataset, types of columns and NaNs.

In [ ]:
data_sea.head()

In [ ]:
data_sea.shape

In [ ]:
# Dropping columns we don't want
data_sea.drop(['Source Data', 'Day', 'Missing'], axis=1, inplace=True) 

In [ ]:
data_sea.dtypes

Types are correct.

In [ ]:
data_sea.isna().sum()

In [ ]:
data_sea.head()

In [ ]:
#Saving clean dataset
data_sea.to_csv('/Users/anna/data/climate-change/datasets/clean_ice.csv', index=False)